Subject: Linux-RT 2.6.25.4-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
Subject: Linux-RT 2.6.25-rt
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
Subject: ftrace fixup patch
Date: Tue, 26 Feb 2008 11:48:47 -0600
From: Clark Williams <williams@redhat.com>
To: Steven Rostedt <srostedt@redhat.com>,        rhel-rt-internal 
<rhel-rt-internal@redhat.com>

-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

Steven,

Attached is a small patch for fixing ftrace compile issues.

Clark
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.8 (GNU/Linux)
Comment: Using GnuPG with Fedora - http://enigmail.mozdev.org

iEYEARECAAYFAkfEUP4ACgkQHyuj/+TTEp07PACcDq1VeUiKhS8tXH9csdpbZc13
qpIAoIquqwObP3CB13WeoqxWoOZE2m/T
=yCp3
-----END PGP SIGNATURE-----


    [ Part 2: "Attached Text" ]

Minor tweaks to fix ftrace compilation problems

Signed-off-by: Clark Williams <williams@redhat.com>

---

 include/linux/ftrace.h |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)


Index: linux-2.6.25-rt1/include/linux/ftrace.h
===================================================================
--- linux-2.6.25-rt1.orig/include/linux/ftrace.h	2008-05-02 22:16:40.000000000 -0400
+++ linux-2.6.25-rt1/include/linux/ftrace.h	2008-05-02 22:16:40.000000000 -0400
@@ -4,7 +4,6 @@
 #ifdef CONFIG_FTRACE
 
 #include <linux/linkage.h>
-#include <linux/ktime.h>
 
 extern int ftrace_enabled;
 extern int
@@ -130,6 +129,9 @@ enum ftrace_event_enum {
 	FTRACE_EVENTS_TIMESTAMP,
 	FTRACE_EVENTS_TASK,
 };
+
+#include <linux/ktime.h>
+
 extern void ftrace_record_event(enum ftrace_event_enum event, ...);
 static inline void ftrace_event_irq(int irq, int user, unsigned long ip)
 {
@@ -179,8 +181,8 @@ static inline void ftrace_event_task(pid
 extern void user_trace_start(void);
 extern void user_trace_stop(void);
 #else
-# define user_trace_start do { } while (0)
-# define user_trace_stop do { } while (0)
+# define user_trace_start() do { } while (0)
+# define user_trace_stop() do { } while (0)
 #endif
 
 #endif /* _LINUX_FTRACE_H */
